### ISSUE TITLE
Classes Not detected. 

### ISSUE THREAD

**Issue by lalitjoshi1989** [ultralytics/yolov5#10759]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I am trying to train my custom dataset using yolov5s. I have annotated my images and I have 17 classes. After the completion of the epochs, the model shows only 11 classes. 1st image shows the results and the second image shows the labels. 

![image](https://user-images.githubusercontent.com/83167348/212255812-8d8073a0-f217-4f68-9315-52897271f131.png)

![image](https://user-images.githubusercontent.com/83167348/212255971-1150edfa-5c7a-43ef-b6de-4263747cc44f.png)

**Comment by mnbrshd** [comment]
Those classes might not be present in the test data and all their instances could be in only the training data. Check the distribution of your data.

**Comment by lalitjoshi1989** [comment]
@mnbrshd Thanks, will try changing the data split. Also, I created the bounding boxes using the polygon technique. But when I am updating the data onto ultralytics it gets converted to boxes. Why so? can you help?

**Comment by mnbrshd** [comment]
@lalitjoshi1989 What is the annotation format of the data when you're annotating? COCO or YOLO?

**Comment by lalitjoshi1989** [comment]
@mnbrshd it's Yolo, JSON format file I am getting after creating the bounding boxes.

**Comment by lalitjoshi1989** [comment]
Also, I am using app.roboflow to create the dataset and ultralytics hub to train the model.

**Comment by mnbrshd** [comment]
YOLO follows "object-class x y width height" format. It is supposed to be like this. If you are trying to do segmentation or if your bounding boxes are rotated, you might need to use the segmentation models whose information is given in the repo. Otherwise, for simple object detection, YOLO will convert your bounding boxes to the  "object-class x y width height" format.

**Comment by glenn-jocher** [comment]
@mnbrshd great to hear that you're using YOLO in JSON format and the Ultralytics hub for model training! YOLO requires the "object-class x y width height" format for bounding box annotations. For segmentation or rotated bounding boxes, there are separate models mentioned in the repo. Feel free to refer to the Ultralytics Docs for more detailed information.
